File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> numSet = new HashSet<Integer>();
        for (int n : nums)
        {
            numSet.add(n);
        }
        int longest = 0;

        for (int num : nums)
        {
            if (!numSet.contains(num-1))
            {
                int length = 0;
                while (numSet.contains(length + num))
                {
                    length++;
                }
                longest = Math.max(length,longest);
            }
        }
        return longest;
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
#include <set>
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::set<int> s;
        for (int x: nums)
        {
            std::cout << x << '\n';
            std::cout << s.count(x) << '\n';

            if (s.count(x) == 1) return true;
            s.insert(x);
        }
        return false;
    }
};
